lib/cdo/aws/cloud_formation.rb

```diff
@@ -48,14 +49,43 @@ def initialize(stack:, log: Logger.new(STDOUT), **options)
       stack_name.gsub!(STACK_NAME_INVALID_REGEX, '-')
     end
 
-    # Validates that the template is valid CloudFormation syntax.
+    # Validates that the template is valid CloudFormation syntax by creating a change set.
     # First prints the rendered template (if `verbose`), then either raises an error (if invalid)
     # or prints the pending stack-resource changes.
     def validate
       stack.dry_run = true
       change_stack
     end
 
+    # Lints the template, similar to validate but simply performs
+    # local static analysis on the generated template
+    def lint
+      stack.dry_run = true
+      template = stack.render
+      if options[:verbose]
+        log.info template.lines.map.with_index(1) {|line, i| format("[%4d] %s", i, line)}.join
+      end
+      Tempfile.create(['template', '.yml']) do |tempfile|
+        tempfile.write(template)
+        tempfile.close
+
+        cfn_lint_installed = system('which cfn-lint > /dev/null 2>&1' || 'where cfn-lint > NUL 2>&1')
```

Comment:
I'm not familiar with `where` or `NUL`; what development environment is this meant to support?

Reply:
Windows, but as I'm looking at this, I'm realizing that it's probably much simpler to just do something like:

`cfn_lint_installed = system('cfn-lint --version > /dev/null 2>&1')`

Reply:
And if they're using WSL, then maybe we don't need to worry about windows quirks...

Reply:
```suggestion
        cfn_lint_installed = system('which cfn-lint > /dev/null 2>&1')
```

Reply:
LGTM!